### DISCUSSION TITLE
Does flux-controller supports running multiple pod?

### DISCUSSION THREAD

**Discussion by dieend** [fluxcd/flux2#1820] [Q&A]
We are encountering delays in reconciliation because we have many Kustomization objects. One possible thought we have is increasing the size of ReplicaSet for Flux Controller's Deployment. Is it supported? Are there any assumptions that Flux only has a single pod running at a time?

If we can scale them up, could it handle the reconciliation delay?

We don't have the exact metrics yet, trying to integrate the metrics with datadog but trying to work around https://github.com/fluxcd/terraform-provider-flux/pull/178

Edit:
It took 10 minutes to fully reconciles all changes at peak load.

**Comment by stefanprodan** [comment]
Scaling up the kustomize-controller reconciler can be done using the `--concurrent` [flag](https://github.com/fluxcd/kustomize-controller/blob/main/main.go#L78). The default is set to 4 Kustomizations to be processed in parallel. You could bump this to a higher number, also make sure to increase the CPU requests and limits, otherwise the CPU limit will throttle the controller.

**Reply by dieend** [reply]
Does `--concurrent` different than number of pod? Is there any estimate how much memory/cpu should be assigned per concurrency?

**Reply by dieend** [reply]
@stefanprodan we tried to increase the number of concurrencies and encountered issue with other pods in the same node running kustomize-controller, because it's timing out pinging Kubernetes API server.

It won't be an issue if we can load-balance the kustomization reconciliation between multiple pods in different nodes.

Can you help us clarify if this is supported? If it isn't, what does it take to support that?
